swirl_lm/example/fire/fire_main.py

```diff
@@ -95,22 +95,57 @@
 - `ignition_scale`: The smoothness factor of the tanh function.
 """
 
-from absl import app
+from absl import app, flags
 from swirl_lm.base import driver
 from swirl_lm.base import parameters as parameters_lib
 from swirl_lm.example.fire import fire
 from swirl_lm.example.shared import wildfire_utils
+from swirl_lm.example.fire import fire_uq
 
 
 def main(_):
+
+  uq_sampler = fire_uq.FireUQSampler()
+  fd_samples, md_samples, ws_samples = uq_sampler.generate_samples()
+  data_dump_prefixes = uq_sampler.generate_data_dump_prefixes(
+    flags.FLAGS.data_dump_prefix
+  )
+
   params = parameters_lib.params_from_config_file_flag()
   fire_utils = wildfire_utils.WildfireUtils(params, None)
   simulation = fire.Fire(fire_utils)
   params.source_update_fn_lib = simulation.source_term_update_fn()
   params.additional_states_update_fn = simulation.additional_states_update_fn
   params.preprocessing_states_update_fn = simulation.pre_simulation_update_fn
   params.postprocessing_states_update_fn = simulation.post_simulation_update_fn
-  driver.solver(simulation.initialization, params)
+  strategy, logical_coordinates = driver.strategy_and_coordinates(params)
+
+  for i in range(uq_sampler.start_id, uq_sampler.end_id):
+    fire_utils.fuel_density = fd_samples[i]
+    fire_utils.moisture_density = md_samples[i]
+    fire_utils.update_wind_speed(ws_samples[i])
+    flags.FLAGS.data_dump_prefix = data_dump_prefixes[i]
+
```

Comment:
Let's add some logging here to make it easier to tell which case and parameters the solver is working on.